exec/vector/src/main/codegen/templates/UnionVectorListWriter.java

```diff
@@ -0,0 +1,212 @@
+import org.apache.drill.exec.vector.complex.writer.FieldWriter;
+
+<@pp.dropOutputFile />
+<@pp.changeOutputFile name="/org/apache/drill/exec/vector/complex/impl/UnionVectorListWriter.java" />
+<#include "/@includes/license.ftl" />
+package org.apache.drill.exec.vector.complex.impl;
+
+<#include "/@includes/vv_imports.ftl" />
+
+import java.util.EnumMap;
+import java.util.Map;
+import java.util.function.Function;
+
+/**
+ * Union vector writer for writing list of union-type values
+ */
+public class UnionVectorListWriter extends UnionVectorWriter {
+
+  private final ListVector listVector;
+  private final UInt4Vector offsets;
+  private int listPosition;
+
+  public UnionVectorListWriter(ListVector listVector, FieldWriter parent) {
+    super(listVector.promoteToUnion(), parent);
+    this.listVector = listVector;
+    this.offsets = listVector.getOffsetVector();
+  }
+
+  //===============================FACTORIES FOR COMPLEX TYPE WRITERS=============================================================
+
+  public UnionVectorWriter union() {
+    return this;
+  }
+
+  public MapWriter map() {
+    return typeWriters.computeIfAbsent(MinorType.MAP, type -> new SingleMapUnionListElementWriter(dataVector.getMap(), null, false));
+  }
+
+  public DictWriter dict() {
+    return typeWriters.computeIfAbsent(MinorType.DICT, type -> new SingleDictUnionListElementWriter(dataVector.getDict(), null, false));
+  }
+
+  public ListWriter list() {
+    return typeWriters.computeIfAbsent(MinorType.LIST, type -> new UnionListUnionElementWriter(dataVector.getList()));
+  }
+
+  //===============================FACTORIES FOR PRIMITIVE LIST ELEMENT WRITERS==============================================================
+<#list vv.types as type>
+  <#list type.minor as minor>
+    <#assign lowerName = minor.class?uncap_first />
+    <#assign upperName = minor.class?upper_case />
+    <#assign capName = minor.class?cap_first />
+    <#if lowerName == "int" >
+      <#assign lowerName = "integer" />
+    </#if>
+    <#if !minor.class?starts_with("Decimal")>
+
+  @Override
+  public ${capName}Writer ${lowerName}() {
+    return typeWriters.computeIfAbsent(MinorType.${upperName}, ${capName}UnionListElementWriter::new);
+  }
+      <#if minor.class == "VarDecimal">
+
+  @Override
+  public ${capName}Writer ${lowerName}(int precision, int scale) {
+    return typeWriters.computeIfAbsent(MinorType.${upperName}, type -> new ${capName}UnionListElementWriter(type, precision, scale));
+  }
+      </#if>
+    </#if>
+  </#list>
+</#list>
+
+  //===============================WRITER's METHODS=====================================================================
+  /**
+   * Since this class is extension of list's data vector writer, index in parent writer is
+   * and index in list vector are now different. Parent writer's index is now based on list
+   * offsets and current writer index point to index of list inside list vector.
```

Comment:
Not sure I follow this sentence: `Parent ...`, especially here `based on list
     * offsets and current`

Reply:
I tried to rewrite the comment. Could you please check whether it's more clear now ?

Reply:
Looks much better.